### ISSUE TITLE
Unable to install basic Yii2 app

### ISSUE THREAD

**Issue by yktoo** [yiisoft/yii2-app-basic#29]
I'm following the [official Yii install guide](http://www.yiiframework.com/doc-2.0/guide-start-installation.html), but composer fails to install dependencies:

```bash

$ composer -vvv create-project --prefer-dist yiisoft/yii2-app-basic basic
Downloading https://packagist.org/packages.json
Writing /home/dk/.composer/cache/repo/https---packagist.org/packages.json into cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/p-provider-2013.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/p-provider-2014.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/p-provider-2015-01.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/p-provider-2015-04.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/p-provider-2015-07.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/p-provider-archived.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/p-provider-latest.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/provider-yiisoft$yii2-app-basic.json from cache
Installing yiisoft/yii2-app-basic (2.0.6)
  - Installing yiisoft/yii2-app-basic (2.0.6)
Reading /home/dk/.composer/cache/files/yiisoft/yii2-app-basic/3159d797b754d99c5d8adf8a492ba7522b467ddd.zip from cache
    Loading from cache
    Extracting archive
Executing command (CWD): unzip 'basic//fb7e30be9dc714613d08a6df78ebac2f' -d '/home/dk/Tmp/vendor/composer/95f46cae' && chmod -R u+w '/home/dk/Tmp/vendor/composer/95f46cae'

Created project in basic
Reading ./composer.json
Loading config file ./composer.json
Failed to initialize global composer: Composer could not find the config file: /home/dk/.composer/composer.json
To initialize a project, please create a composer.json file as described in the https://getcomposer.org/ "Getting Started" section
Loading composer repositories with package information
Downloading https://packagist.org/packages.json
Writing /home/dk/.composer/cache/repo/https---packagist.org/packages.json into cache
Installing dependencies (including require-dev)
Reading /home/dk/.composer/cache/repo/https---packagist.org/p-provider-2013.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/p-provider-2014.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/p-provider-2015-01.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/p-provider-2015-04.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/p-provider-2015-07.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/p-provider-archived.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/p-provider-latest.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/provider-yiisoft$yii2-app-basic.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/provider-yiisoft$yii2.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/provider-yiisoft$yii2-bootstrap.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/provider-yiisoft$yii2-swiftmailer.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/provider-yiisoft$yii2-composer.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/provider-ezyang$htmlpurifier.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/provider-cebe$markdown.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/provider-bower-asset$jquery.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/provider-bower-asset$jquery.inputmask.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/provider-bower-asset$punycode.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/provider-bower-asset$yii2-pjax.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/provider-bower-asset$bootstrap.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/provider-swiftmailer$swiftmailer.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/provider-yiisoft$yii2-codeception.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/provider-yiisoft$yii2-debug.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/provider-yiisoft$yii2-gii.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/provider-phpspec$php-diff.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/provider-yiisoft$yii2-faker.json from cache
Reading /home/dk/.composer/cache/repo/https---packagist.org/provider-fzaninotto$faker.json from cache
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - yiisoft/yii2 2.0.6 requires bower-asset/jquery 2.1.*@stable | 1.11.*@stable -> no matching package found.
    - yiisoft/yii2 2.0.5 requires bower-asset/jquery 2.1.*@stable | 1.11.*@stable -> no matching package found.
    - Installation request for yiisoft/yii2 >=2.0.5 -> satisfiable by yiisoft/yii2[2.0.5, 2.0.6].

Potential causes:
 - A typo in the package name
 - The package is not available in a stable-enough version according to your minimum-stability setting
   see <https://groups.google.com/d/topic/composer-dev/_g3ASeIFlrc/discussion> for more details.

Read <https://getcomposer.org/doc/articles/troubleshooting.md> for further common problems.

```

**Comment by onedevlink** [comment]
run `composer global require "fxp/composer-asset-plugin:~1.0.3"` first

**Comment by yktoo** [comment]
I've installed the asset plugin, thanks!

But the result didn't change, see the log here: http://pastebin.com/hNn1sAMY (for an unknown reason GitHub forbad me to attach it here).

**Comment by onedevlink** [comment]
Install guide is correct. Problem in `composer-asset-plugin`. Try purge composer, install new one, install `composer-asset-plugin:~1.0.3` and then create yii2 app basic project.

**Comment by yktoo** [comment]
Okay I did as proposed and got the following after about one hour of running composer:

```
[Composer\Repository\RepositorySecurityException]                                                                                                                            
  The contents of http://packagist.org/p/yiisoft/yii2-gii%244550c9347758a016e2b98f88de8a5d5dc4e4f4e4f7d47e498c1d66a360d9d397.json do not match its signature. This should ind  
  icate a man-in-the-middle attack. Try running composer again and report this if you think it is a mistake. 
```

I'm giving up at this point. I've been trying to get just a sample app for two (!) days and failed.

**Comment by cebe** [comment]
What kind of internet connection are you using?

**Comment by yktoo** [comment]
@cebe My Internet connection is fine (it's 100 Mbps fiber-optic), but nonetheless it's been such a pain to get the sample app. Finally I've managed to get it by specifying the exact version (as given on the [Download](http://www.yiiframework.com/download/) page):

```bash
php composer.phar create-project yiisoft/yii2-app-basic basic 2.0.4
```

All previous attempts of running `composer create-project --prefer-dist yiisoft/yii2-app-basic basic` failed for some totally obscure reason. I suspect part of the problem was that GitHub returns some malformed JSON when the user is over his quota, and after that a lot of things go wrong. The install guide also doesn't provide any details on how to resume the failed dep. update process. And I believe, the first thing to do after installing composer must be [obtaining a token](https://github.com/settings/tokens) and installing it:

```bash
cd ~/.composer && composer config github-oauth.github.com <your_github_token_here>
```

Otherwise you'll never finish the update. Anyway, the install guide warns you that it has nothing to do with composer's quirks, yet it'd be nice to at least provide solutions some common tasks (like resuming the update).

**Comment by cebe** [comment]
Maybe your internet connection was too fast for github then :)

Will check the guide and add a note for this.

**Comment by Artyom1377** [comment]
$ composer create-project --prefer-dist yiisoft/yii2-app-basic yii_task
Installing yiisoft/yii2-app-basic (2.0.12)
  - Installing yiisoft/yii2-app-basic (2.0.12): Downloading (failed)
Downloading (failed)
Downloading (failed)    Failed to download yiisoft/yii2-app-basic from dist: The "https://api.github.com/repos/yiisoft/yii2-app-basic/zipball/591a3ca78caffc9e381f714da99a093a618d4169" file could not be downloaded: SSL operation failed with code 1. OpenSSL Error messages:
error:14077419:SSL routines:SSL23_GET_SERVER_HELLO:tlsv1 alert access denied
Failed to enable crypto
failed to open stream: operation failed
    Now trying to download from source
  - Installing yiisoft/yii2-app-basic (2.0.12): Cloning 591a3ca78c from cache
Created project in yii_task
Loading composer repositories with package information
Updating dependencies (including require-dev)


  [Composer\Repository\InvalidRepositoryException]
  No valid bower.json was found in any branch or tag of https://github.com/jq
  uery/jquery-dist.git, could not load a package from it.

how can i solve it?

**Comment by schmunk42** [comment]
What does `composer diagnose` say?

**Comment by Artyom1377** [comment]
$ composer diagnose
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com oauth access: FAIL
[Composer\Downloader\TransportException] The "https://api.github.com/" file could not be downloaded: SSL operation failed with code 1. OpenSSL Error messages:
error:14077419:SSL routines:SSL23_GET_SERVER_HELLO:tlsv1 alert access denied
Failed to enable crypto
failed to open stream: operation failed
Checking disk free space: OK
Checking pubkeys:
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK

**Comment by schmunk42** [comment]
Looks like some issue with the SSL module of PHP, see also:

- https://stackoverflow.com/questions/26148701/file-get-contents-ssl-operation-failed-with-code-1-and-more
- https://stackoverflow.com/questions/15166950/unable-to-establish-ssl-connection-how-do-i-fix-my-ssl-cert/15168180#15168180

**Comment by Artyom1377** [comment]
Thank you for help, but it doesn't work for me
